config/configtls/configtls.go

```diff
@@ -153,6 +153,24 @@ func (r *certReloader) GetCertificate() (*tls.Certificate, error) {
 	return r.cert, nil
 }
 
+func (c TLSSetting) Validate() error {
+	minTLS, err := convertVersion(c.MinVersion, defaultMinTLSVersion)
+	if err != nil {
+		return fmt.Errorf("invalid TLS min_version: %w", err)
+	}
+
+	maxTLS, err := convertVersion(c.MaxVersion, defaultMaxTLSVersion)
+	if err != nil {
+		return fmt.Errorf("invalid TLS max_version: %w", err)
+	}
+
+	if maxTLS < minTLS && maxTLS != defaultMaxTLSVersion {
```

Comment:
What if the user sets `max_version` lower than `min_version` but the same as the default value? Looks like it should be checked before falling back to the defaults

Reply:
Currently `defaultMaxTLSVersion` is set to 0 and user is not able to provide such value. `convertVersion` would return an error, because `"0"` is not present in `tlsVersions` [map](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/configtls.go#L410).
There are cases when `maxTLS` returned by `convertVersion` is lower than `minTLS`, e.g. user provides value for `min_version`, but omits `max_version`. In such case `maxTLS == 0` and `minVersion` is a non-zero value (greater than zero). When `0` is provided to tls.Config, the max supported version is used by `tls` package.

I guess there is a risk when `defaultMaxVersion` is set to a different value than `0`. Maybe it would be a better idea to explicitly compare `maxTLS` with `0`?
```
if maxTLS < minTLS && maxTLS != 0 {`
```

Reply:
I believe `if maxTLS < minTLS && maxTLS != 0 {`  is cleaner. I would even remove `defaultMaxTLSVersion` completely and use zero value instead, but that's out of the scope probably